plugins/appwrite/appwrite.go

```diff
@@ -0,0 +1,39 @@
+package appwrite
+
+import (
+	"github.com/1Password/shell-plugins/sdk"
+	"github.com/1Password/shell-plugins/sdk/needsauth"
+	"github.com/1Password/shell-plugins/sdk/schema"
+	"github.com/1Password/shell-plugins/sdk/schema/credname"
+)
+
+func AppwriteCLI() schema.Executable {
+	return schema.Executable{
+		Name:    "Appwrite CLI",
+		Runs:    []string{"appwrite"},
+		DocsURL: sdk.URL("https://appwrite.io/docs/command-line-ci"),
+		NeedsAuth: needsauth.IfAll(
+			needsauth.NotForHelpOrVersion(),
+			needsauth.NotWithoutArgs(),
+			needsauth.NotWhenContainsArgs("client"),
+			needsauth.NotWhenContainsArgs("login"),
+			needsauth.NotWhenContainsArgs("logout"),
+			needsauth.NotForExactArgs("deploy"),
+			needsauth.NotWhenContainsArgs("projects"),
+			needsauth.NotForExactArgs("storage"),
+			needsauth.NotForExactArgs("teams"),
+			needsauth.NotForExactArgs("users"),
+			needsauth.NotForExactArgs("account"),
+			needsauth.NotForExactArgs("avatars"),
```

Comment:
These looks like commands that would be scoped to a specific account. 

Is there any documentation you could point me to about these commands? I wonder why they would not require auth.

Reply:
@hculea without any argument these commands behave as if they were called with --help argument. So they just give you the list of available subcommands and options. Also, these are project specific commands. So if you run `appwrite users list` from the project folder (can only be run from project folder which contains the `appwrite.json` config file), you'll get the list of all users of that project.

Here is the [CLI documentation](https://appwrite.io/docs/command-line-commands) (they don't say the above in the docs, but all the examples given there are with arguments). I think @arunsathiya has verified this.

Reply:
From my functional testing: The project token that we provision seems to work for `users list` and `teams list`. I'd imagine that it works for the rest of the subcommands for `users` and `teams`.

However, `account get` (one of the many subcommands under `account`) does not work with the project token, rather requires account scope. I think that's expecting a combination of username and password?

```
➜  ~ appwrite account get
#########################################################################
# WARNING: 'appwrite' is not from the official registry.                #
# Only proceed if you are the developer of 'appwrite'.                  #
# Otherwise, delete the file at /Users/arun/.op/plugins/local/appwrite. #
#########################################################################
✗ Error app.6495df658da7c618cab3@service.cloud.appwrite.io (role: applications) missing scope (account)
```

Not too sure how `avatars`' subcommands work though. 

>without any argument these commands behave as if they were called with --help argument

This matches my testing too. 👍🏼

Reply:
`account` commands work with currently logged in user (requires a session). `users` is a more appropriate command from server's perspective. `account` is for impersonating a user and acting in their behalf using their JWT token.

`avatars` offers below subcommands:
>getBrowser
getCreditCard
getFavicon
getFlag
getImage
getInitials
getQR

These are some utility functions for getting images/icons for stuffs. e.g. getBrowser is for getting browser icon, getFlag for getting a country's flag icon etc.

Reply:
So if account doesn't work with the API KEY we provision anyway (it only works with a JWT), we should make sure we also exclude this command from out auth scheme as well: `needsauth.NotWhenContainsArgs("account"),`.

As a bit of a background, where does that JWT come from? We should make sure this doesn't get placed somewhere on disk in plain-text.

Reply:
@AndyTitu The JWT comes when you use a Login/Password to login to the CLI (which we can't use with 1Passsword as this is an interactive process). Also, in this flow the JWT actually gets written to the same config file (~/.appwrite/prefs.json) on disk, again there won't be any JWT if we're using the CI mode (which we're doing with this integration).

You're right that the `account` commands don't work with an API KEY. Please let me know if I need to make this change `needsauth.NotWhenContainsArgs("account")`.

Thanks,
Rajeev

Reply:
Yes, I think that change makes sense.

Sorry for the delayed reply!

Reply:
No problem @accraw. Don't think this will be merged so it is okay. :-)